### PR TITLE
Fix karma, readme updates, combine source to one file

### LINKED ISSUE #78: Improve type definitions for decode
Thinking abstractly about `decode` I'd expect a function signature such as:

```typescript
export function decode(input: Buffer): NestedBufferList
```

where `NestedBufferList` could be any nested composition of buffers and arrays such as `Buffer`, or `Buffer[]`, or `[Buffer, Buffer[]]`, etc. (haven't tried expressing this in TS).

Currently the `decode` method has the following definitions:

https://github.com/ethereumjs/rlp/blob/f06a96fa061127d8b85c45bd1dfae1b067426c0f/src/index.ts#L60-L63

1. First definition takes `Buffer` and returns a `Buffer`. Not sure why, it could very well be that the decoding is a `NestedBufferList`. This was also the problem stated in #77
2. Second def. takes a `Buffer[]`. Is this to support decoding multiple serialized items simultaneously? the code doesn't seem to do that. I suggest we remove this
3. The last two take `Input` as argument. I imagine this is for backwards-compatibility. I suggest we tighten the API and only accept a `Buffer`

On the `stream` flag: we could keep it, or add a new function `decodeStreaming`.

### PR DESCRIPTION
**Pull request by ryanio**
This PR updates the karma-typescript dependency to a built version from master using gitpkg for the unreleased [fix](https://github.com/monounity/karma-typescript/pull/502).

It also includes some refactoring and readme updates.

### CODE REVIEW

**Comment by paulmillr** [comment]
I know using 3 files instead of 1 is much more readable, just not sure about ESM support - or Deno support. 

I have been making my stuff work with those, and there are some things that you need to take into account. 

For example, in Deno you can't import .js files, and you need to specify extensions in imports. In ESM, you cannot import .ts files.

My suggestion would be to move everything into one file and place it in root. After that, after the release, we could work on .esm tsconfig and Deno package. I could help here

**Comment by paulmillr** [comment]
I have worked around "many files" thing in bls12-381 by using export maps but we definitely don't want to force Deno users to do that (a separate .json file)

**Comment by ryanio** [comment]
@paulmillr sounds good, thanks for the feedback, we already had 2 files (index, types) so I thought a third for utils would be harmless (the improved readability is great), but I am also in favor of broader ecosystem support, so whatever we need to do here - if it is one file then that's fine by me too.

Appreciate your help with esm and Deno insights, we can probably bring these learnings over to our monorepo as well.

**Comment by ryanio** [comment]
For the karma-typescript fix, I wrote some of my findings in this comment: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1602#issuecomment-1008222864

**Comment by ryanio** [comment]
@talentlessguy has helped us with Deno compatibility in the past, let us know how this PR is looking

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [12.x, 13.x, 14.x, 16.x]
+        node-version: [12, 13, 14, 16]
```
Comment (paulmillr) [inline .github/workflows/build.yml:15]: Let's remove v13, it has not been supported for many years.

**Review by paulmillr — CHANGES_REQUESTED** [review]
Looks great to me!

**Review by paulmillr — APPROVED** [review]

**Comment by ryanio** [comment]
thanks!

**Pull request merged**